### PR TITLE
Algolia: Log timing of all requests on the server

### PR DESCRIPTION
**Pull request by danielbeardsley**
Found a way to get it done!

First step is to measure, second will be to optimize.

Connect #1641

### CODE REVIEW

**Review by danielcliu-ifixit — APPROVED** [review]
CR :v:

**Comment by danielbeardsley** [comment]
dev_block :+1: 

On pages that end up querying algolia mutliple times (once for the results, once for some sub-device info) this logs two entries. I'm going to try to only log during the primary result-set gathering phase.

image

**Comment by danielbeardsley** [comment]
un_dev_block :+1: 

Did the thing!

**Comment by danielbeardsley** [comment]
image

image

QA :+1:

File: frontend/helpers/algolia-helpers.ts [hunk 1]
```diff
@@ -1,31 +1,51 @@
...
+   logContextName: string,
+   requester: Requester
+): Requester {
+   let count = 0;
    return {
-      send: withTiming('algolia.request', requester.send),
+      send(request: Request): Promise<Response> {
+         count++;
+         const eventSuffix = count > 1 ? count : '';
+         const eventName = logContextName + eventSuffix;
```
Comment (danielcliu-ifixit) [inline frontend/helpers/algolia-helpers.ts:46]: do we want a space or anything?

**Review by danielcliu-ifixit — APPROVED** [review]
CR :v:

File: frontend/helpers/algolia-helpers.ts (hunk 1 above)
Comment (danielbeardsley) [reply frontend/helpers/algolia-helpers.ts:46]: We could have done a `-2` but not a space, but I opted to keep it simple.

**Pull request merged**